### PR TITLE
Add changelog for beta 4

### PR DESCRIPTION
**Pull request by dotNomad**
This PR adds details for beta to the CHANGELOG

It updates the root level changelog, and the one included in the VSCode extension.

## Intent

Part of #1967

## Type of Change

- - [ ] Bug Fix 
- - [ ] New Feature 
- - [ ] Breaking Change 
- - [x] Documentation 
- - [ ] Refactor 
- - [ ] Tooling 

## Approach

I left out code refactors like we normally do, but also didn't include fixes to bugs / regressions caused during Beta 4 development.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,37 @@ All notable changes to this project will be documented in this file.
...
+### Added
+
+- Added the support for recursively scanning workspace directories so the
+  extension can manage multiple projects at once (#1843, #1844, #1845, #1849,
+  #1850, #1851, #1852, #1853, #1854, #1855, #1944)
+- Added a publish button to the editor view when an entrypoint is the active
+  file to allow for easier deployment of the entrypoint (#1846, #1848, #2003)
+- Added progress bars for asynchronous actions in each of the extension's views
+  (#1958)
+- Retained deployment errors in pre-Deployment files (#2004)
```
Comment (mmarchetti) [inline CHANGELOG.md:19]: This was already the current behavior before 1.1.4.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,37 @@ All notable changes to this project will be documented in this file.
...
 
+## [1.1.4]
+
+### Added
+
+- Added the support for recursively scanning workspace directories so the
+  extension can manage multiple projects at once (#1843, #1844, #1845, #1849,
+  #1850, #1851, #1852, #1853, #1854, #1855, #1944)
+- Added a publish button to the editor view when an entrypoint is the active
+  file to allow for easier deployment of the entrypoint (#1846, #1848, #2003)
```
Comment (mmarchetti) [inline CHANGELOG.md:16]: Similarly, this could reference the epic (#1846).

File: CHANGELOG.md (hunk 1 above)
Comment (dotNomad) [reply CHANGELOG.md:19]: Awesome I'll remove this line.

File: CHANGELOG.md (hunk 2 above)
Comment (dotNomad) [reply CHANGELOG.md]: Updated in https://github.com/posit-dev/publisher/pull/2015/commits/c6b9d10f151ae6cf57ab1d51f28ff6d1f2fa67ed

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,37 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [1.1.4]
+
+### Added
+
+- Added the support for recursively scanning workspace directories so the
+  extension can manage multiple projects at once (#1843, #1844, #1845, #1849,
```
[the remark this answers is not in the document]
Comment (dotNomad) [reply CHANGELOG.md]: Updated in https://github.com/posit-dev/publisher/pull/2015/commits/c6b9d10f151ae6cf57ab1d51f28ff6d1f2fa67ed

File: CHANGELOG.md (hunk 1 above)
Comment (dotNomad) [reply CHANGELOG.md]: Removed in https://github.com/posit-dev/publisher/pull/2015/commits/941fa534298165b6926c693f2aef76c04a0b85db

**Review by sagerb — APPROVED** [review]
LGTM! Thanks for doing this!

**Pull request merged**